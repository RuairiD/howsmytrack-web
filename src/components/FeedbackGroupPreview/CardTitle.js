import React from "react";

import { Badge, Typography } from "antd";
import { A, Div } from "lemon-reset";

const CardTitle = ({ feedbackGroupId, name, unreadReplies }) => (
    <A href={`/group/${feedbackGroupId}`}>
        <Div style={{ display: "flex" }}>
            <Typography.Title level={4} style={{ marginRight: "auto", marginBottom: "auto", marginTop: "auto" }}>
                {name}
            </Typography.Title>
            <Badge
                count={unreadReplies}
                style={{ marginLeft: "auto", marginBottom: "auto", marginTop: "auto" }}
            />
        </Div>
    </A>
);

export default CardTitle;
