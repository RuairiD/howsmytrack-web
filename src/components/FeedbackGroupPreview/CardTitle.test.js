import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import CardTitle from "./CardTitle";

describe("CardTitle", () => {
    it("renders the header for a FeedbackGroupPreview with a formatted URL and badge for unread replies", () => {
        const wrapper = shallow(
            <CardTitle
                feedbackGroupId={1901}
                name="name"
                unreadReplies={2}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
