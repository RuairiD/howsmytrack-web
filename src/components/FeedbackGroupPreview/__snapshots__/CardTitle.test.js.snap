// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTitle renders the header for a FeedbackGroupPreview with a formatted URL and badge for unread replies 1`] = `
<A
  className=""
  href="/group/1901"
>
  <Div
    className=""
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Title
      level={4}
      style={
        Object {
          "marginBottom": "auto",
          "marginRight": "auto",
          "marginTop": "auto",
        }
      }
    >
      name
    </Title>
    <Badge
      count={2}
      dot={false}
      overflowCount={99}
      showZero={false}
      style={
        Object {
          "marginBottom": "auto",
          "marginLeft": "auto",
          "marginTop": "auto",
        }
      }
    />
  </Div>
</A>
`;
